server/main.go

```diff
@@ -98,13 +99,15 @@ func NameHandler(w http.ResponseWriter, r *http.Request) {
 
 	url := fmt.Sprintf("https://buildpack-registry.s3.amazonaws.com/buildpacks/%s.tgz", id)
 	log.Infof("at=download file=%s url=%s", shimmedBuildpack, url)
-	cmd := fmt.Sprintf(`curl --retry 3 --silent --location "%s" | tar xzm -C %s`, url, target_dir)
-
-	_, err = exec.Command("bash", "-c", cmd).Output()
+	bp, err = downloadBuildpack(url)
+	handlePanic(err)
+	tar := fmt.Sprintf(`tar xzf %s -C %s`, bp, target_dir)
+	_, err = exec.Command("bash", "-c", tar).Output()
 	handlePanic(err)
+	handlePanic(os.Remove(bp))
```

Comment:
our tar is also including the tgz file into the target dir, I don't know why it didn't do that in the `curl ... | tar -C` command but this was the quickest and easiest way to get it out. 

If someone's tar foo is better than mine I'd like to remove this but it was becoming a time sink.